subprojects/version-control/src/main/java/org/gradle/vcs/internal/DefaultVersionControlSystemFactory.java

```diff
@@ -16,17 +16,66 @@
 
 package org.gradle.vcs.internal;
 
+import org.gradle.cache.CacheRepository;
+import org.gradle.cache.FileLockManager;
+import org.gradle.cache.PersistentCache;
 import org.gradle.vcs.VersionControlSpec;
 import org.gradle.vcs.VersionControlSystem;
+import org.gradle.vcs.VersionRef;
+import org.gradle.vcs.git.GitVersionControlSpec;
 import org.gradle.vcs.git.internal.GitVersionControlSystem;
 
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.gradle.cache.internal.filelock.LockOptionsBuilder.mode;
+
 public class DefaultVersionControlSystemFactory implements VersionControlSystemFactory {
+    private final CacheRepository cacheRepository;
+    private final Map<Class<? extends VersionControlSpec>, VersionControlSystem> versionControlSystems;
+
+    DefaultVersionControlSystemFactory(CacheRepository cacheRepository) {
+        this.cacheRepository = cacheRepository;
+        // TODO: Move this map to a registry and inject the registry.
+        this.versionControlSystems = new HashMap<Class<? extends VersionControlSpec>, VersionControlSystem>();
+        versionControlSystems.put(DirectoryRepository.class, new SimpleVersionControlSystem());
+        versionControlSystems.put(GitVersionControlSpec.class, new GitVersionControlSystem());
```

Comment:
I think we may want to eventually allow services to be injected into these and we may want to create a new instance for every spec (see my comment below).